wolfcrypt/src/ed25519.c

```diff
@@ -589,13 +668,18 @@ int wc_ed25519_init_ex(ed25519_key* key, void* heap, int devId)
 #else
     (void)devId;
 #endif
-    (void)heap; /* if needed for XMALLOC/XFREE in future */
+    key->heap = heap;
 
 #ifndef FREESCALE_LTC_ECC
     fe_init();
 #endif
 
-    return 0;
+    ret = wc_InitSha512_ex(&key->sha, key->heap, INVALID_DEVID);
```

Comment:
The `devId` should also be passed in here ideally.

Reply:
done